NFC driver adapt for Matter v1.4

interfer opened the pull request:
Introduce NFC driver for ST25DV module for Telink boards.
Tested on B92 with Matter lighting-app example app.

Currently NFC sample in this PR is not fully functional as some of modules used there are under inapropriate license.
But in minimal functionality (NDEF) it can work fine in Matter onboarding payload via NFC.

ST drivers are in submodule of external repo due to BSD-3-Clause licence policy.

andriy-bilynskyy commented:
The following west manifest projects have changed revision in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| nfc_st25dv 🆕 | N/A (Added) | https://github.com/telink-semi/nfc_st25dv/commit/8854cf1e33f60e85a96e84bf35e442660883324f (`main`)| N/A |

⛔ **DNM label due to: 1 added project**

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

s07641069 committed:
drivers: nfc: Add NFC driver

Copied st25xvxxkc driver from https://github.com/STMicroelectronics/x-cube-nfc7 (BSD-3-Clause)
https://github.com/STMicroelectronics/x-cube-nfc7?tab=License-1-ov-file

Signed-off-by: Alex Tsitsiura <s07641069@gmail.com>

drivers: NFC build fix, made as submodule

As APPLICATION is now deprecated in Zephyr, deffered-init
introduced instead for init from app.
Usage example:
```c
const struct device *nfc_dev = DEVICE_DT_GET(DT_NODELABEL(st25dvxxkc));

    if (!device_is_ready(nfc_dev)) {
        int ret = device_init(nfc_dev);
        if (ret != 0) {
            printf("Failed to initialize NFC device: %d", ret);
            return -1;
        }
    }
```

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

drivers: fixed I2C address for ST25DV

Sample samples/subsys/nfc/ confirmed to be working.
NOTE: should include 4k7 Ohm pullups for SDA,SCL.
```
*** Booting Zephyr OS build v3.1.0-rc1-46872-gd9aff28ed307 ***
Starting NFC Text Record example
Device = ok: st25dvxxkc
[19:52:50.875,343] <dbg> st25dvxxkc: _st25dvxxkc_init: init
[19:52:50.875,375] <dbg> st25dvxxkc: _st25dvxxkc_init: init OK
NFC configuration done (0)
```

Also added DT overlay to work with B92, tested as well

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

subsys: nfc: removed non-Apache lisensed modules

All modules under subsys/nfc (ndef, t2t, t4t and tnep) were
licensed as Nordic specific and cannot be used with other than
nRF IC.
This commit breaks functionality of NFC sample (samples/subsys/nfc/).
For Matter we only need NDEF encode so now it's handled in there in:
src/platform/telink/NFCOnboardingPayloadManagerImpl.cpp

Still those modules need to be replaced with some compatible license ones.
Only then NFC sample can be fixed.

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

drivers: migrated NFC driver to submodule

Removed everything with license conflict
to `modules/lib/nfc_st25dv`.
It's all BSD or MIT so no legal issues.
Submodule is introduced only because Zephyr in-tree modules
can be only Apache 2.0.

Sample still broken but with few mods can be used for linkage testing.

Signed-off-by: Dmytro Huz <diman1436@gmail.com>
